client/components/checkout.js

```diff
@@ -29,70 +29,67 @@ class Checkout extends Component {
     this.handleReview = this.handleReview.bind(this)
     this.getLastFour = this.getLastFour.bind(this)
     this.handleChange = this.handleChange.bind(this)
-    this.handleConfirmation = this.handleConfirmation.bind(this)
+    this.handleTopLevelChange = this.handleTopLevelChange.bind(this)
   }
   componentDidMount() {
+    this.setUserState()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.user !== this.props.user) {
+      this.setUserState()
+    }
+  }
+
+  setUserState() {
+    const {user: {googleId: _, ...restOfUser}} = this.props
     this.setState({
-      firstName: this.props.user.firstName || '',
-      lastName: this.props.user.lastName || '',
-      email: this.props.user.email || '',
-      streetAddress: this.props.user.streetAddress || '',
-      city: this.props.user.city || '',
-      state: this.props.user.state || '',
-      zipCode: this.props.user.zipCode || '',
+      user: restOfUser || {},
       cardNumber: '',
       expirationDate: '',
-      cvv: '',
-      showReview: false,
-      showConfirmation: false
+      cvv: ''
     })
   }
-  handleChange(evt) {
-    evt.preventDefault()
-    this.setState({[evt.target.name]: evt.target.value})
+
+  handleChange() {
+    this.setState({
+      user: {...this.state.user, [event.target.name]: event.target.value}
+    })
   }
-  //update user
-  async handleSubmit(evt) {
-    evt.preventDefault()
-    console.log('handle submit')
+
+  handleTopLevelChange(event) {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+  }
+
+  handleSubmit(event) {
+    event.preventDefault()
+
     try {
```

Comment:
Can remove try catch since we're setting the state on the component without any AJAX requests